runtime/lib/descriptor/keyboard-descriptor.js

```diff
@@ -83,8 +83,12 @@ Object.assign(KeyboardDescriptor.prototype,
           return Promise.reject(new Error('Expect a string on second argument of keyboard.preventDefaults.'))
         }
         var events = Object.keys(this.interests)
-        if (event != null && events.indexOf(event) >= 0) {
-          events = [ event ]
+        if (event != null) {
+          if (events.indexOf(event) >= 0) {
+            events = [ event ]
+          } else {
+            return Promise.reject(new Error(`Unexpected keyboard event: ${event}.`))
```

Comment:
I'd suggest an early assertion pattern to reduce nesting levels, e.g.

```js
if (event != null) {
  if (events.indexOf(event) < 0) {
    return Promise.reject(new Error(`Unexpected keyboard event: ${event}.`))
  }
  events = [ event ]
}
```

Reply:
Merge some logic branches together too.